ion-schema/src/model/constraints/timestamp_precision.rs

```diff
@@ -0,0 +1,300 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::internal_traits::{
+    LoaderContext, ReadFromIsl, ValidateInternal, ValidationContext, WriteAsIsl, WriteContext,
+};
+use crate::ion_extension::ElementExtensions;
+use crate::model::constraints::ConstraintName;
+use crate::model::ranges::IonSchemaRange;
+use crate::model::TypeDefinitionBuilder;
+use crate::result::{invalid_schema_error, IonSchemaResult};
+use crate::{IonSchemaElement, IslVersion, ViolationInfo, ViolationRecorder};
+use ion_rs::{Element, IonResult, TimestampPrecision as TSPrecision, ValueWriter, WriteAsIon};
+use std::fmt::{Debug, Display, Formatter};
+use std::ops::ControlFlow;
+
+// For brevity within this module.
+use TimestampPrecisionValue::*;
+// Associated constants still need to be qualified, even with a * import.
+use TimestampPrecisionValue as Tpv;
```

Comment:
I'd be ok with either, but having both seems unusual?

Reply:
Well, I like being able to use the enum variants without qualification (which is why the first one is there). If we're getting rid of one of them, I'd probably get rid of the second one.